File: Exo/ExoPHP/Rectangle/Rectangle.php
<?php
require('Polygone.php');
class Rectangle extends Polygone{

/********************Variable*******************/


/********************Accesseur*******************/


/********************Construct*******************/



/********************Methode*******************/

    public function perimetre()  {
        return 2*($this->_largeur+$this->_longueur);
    }
    public function aire()  {
        return $this->_largeur*$this->_longueur;
    }
    public function estCarre()  {
        if ($this->_largeur == $this->_longueur) {
            return true ;
        }   
    }
    public function afficherTriangle()  {
        echo $this; 
    }
    public function __toString(){
        $retour = "Longueur : ".$this->_longueur."- Largeur : ".$this->_largeur." - Périmètre : ".$this->perimetre()." - Aire : ".$this->aire() ;
        if ($this->estCarre()) {
            $retour = $retour." Il s’agit d’un carré ";
        }else{
            $retour = $retour." Il ne s’agit pas d’un carré ";
        }
        return $retour;
    }
}

File: Exo/exoCs/Eco_03/ExoClasse/Exo_Class_04/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Exo_Class_04
{
    internal class Rectangle
    {
        public int Longueur { get; set; }
        public int Largeur { get; set; }

        public Rectangle(int longueur, int largeur)
        {
            Longueur = longueur;
            Largeur = largeur;
        }

        public int Perimetre() 
        {
            return (Longueur+Largeur)*2;
        }
        public int Aire()
        {
            return Longueur * Largeur;
        }
        public bool EstCarre()
        {
            return Longueur == Largeur;
        }

        public String AfficherRectangle()
        {
            return "Longueur : " + Longueur + " - Largeur : " + Largeur + " - Périmètre : " + Perimetre() + " - Aire : " + Aire() + (EstCarre()? " - Il s’agit d’un carré": " - Il ne s’agit pas d’un carré");
        } 
    }
}
